js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/PartitionSegmentWithPopover.tsx

```diff
@@ -34,45 +38,72 @@ const statusToColors = (status: AssetConditionEvaluationStatus) => {
   }
 };
 
+type AssetSusbsetWithoutTypenames = {
+  subsetValue: Omit<AssetSubsetValue, '__typename' | 'boolValue'>;
+};
+
 interface Props {
   description: string;
   status: AssetConditionEvaluationStatus;
-  subset: AssetSubset | null;
-  width: number;
+  subset: AssetSusbsetWithoutTypenames | null;
+  selectPartition: (partitionKey: string | null) => void;
 }
+export const PartitionSegmentWithPopover = ({
+  description,
+  selectPartition,
+  status,
+  subset,
+}: Props) => {
+  const intent = React.useMemo(() => {
+    switch (status) {
+      case AssetConditionEvaluationStatus.FALSE:
+        return 'warning' as const;
+      case AssetConditionEvaluationStatus.SKIPPED:
+        return undefined;
+      case AssetConditionEvaluationStatus.TRUE:
+        return 'success' as const;
+    }
+  }, [status]);
 
-export const PartitionSegmentWithPopover = ({description, width, status, subset}: Props) => {
-  const {color, hoverColor} = React.useMemo(() => statusToColors(status), [status]);
-  const segment = <PartitionSegment $color={color} $hoverColor={hoverColor} $width={width} />;
   if (!subset) {
-    return segment;
+    return null;
   }
 
+  const count = subset.subsetValue.partitionKeys?.length || 0;
+
   return (
-    <SegmentContainer $width={width}>
-      <Popover
-        interactionKind="hover"
-        placement="bottom"
-        hoverOpenDelay={50}
-        hoverCloseDelay={50}
-        content={<PartitionSubsetList description={description} status={status} subset={subset} />}
-      >
-        {segment}
-      </Popover>
-    </SegmentContainer>
+    <Popover
+      interactionKind="hover"
+      placement="bottom"
+      hoverOpenDelay={50}
+      hoverCloseDelay={50}
+      content={
+        <PartitionSubsetList
+          description={description}
+          status={status}
+          subset={subset}
+          selectPartition={selectPartition}
+        />
+      }
+    >
+      <Tag intent={intent}>
+        {numberFormatter.format(count)} {status.charAt(0) + status.toLowerCase().slice(1)}
```

Comment:
Not really a fan of slicing strings for copy like this. Maybe put it in a span and apply `text-transform`?

Reply:
does text-transform have an option to only capitalize the first letter?

Reply:
`text-transform: capitalize` seems correct. [[link](https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform#example_using_capitalize_general)]

Reply:
It left "LONDON" complete capitalized so it's not correct.